pkg/injector/pod_patch.go

```diff
@@ -155,7 +155,7 @@ func (i *injector) getPodPatchOperations(ar *v1.AdmissionReview,
 		value = []corev1.Container{*sidecarContainer}
 	} else {
 		envPatchOps = addDaprEnvVarsToContainers(pod.Spec.Containers)
-		path = "/spec/containers/-"
+		path = "/spec/containers/0"
```

Comment:
I suspect this is what's failing the tests - when I tried this in my PR, it changes the container order and this messed up arguments and other things.

I suggest you try to build your own image of the sidecar injector with this change, deploy a sample app and then do `kubectl get pod <pod-name> -o yaml` to see the container settings and compare them with a properly injected app.

Reply:
@skyao do you need any help here?

Reply:
> I suspect this is what's failing the tests - when I tried this in my PR, it changes the container order and this messed up arguments and other things.

hi, Yaron, you are right. I have checked the yaml of app pod after making daprd to the first container. 

**The env content is not correct!**

The right env setting for daprd container should be (I changed yaml to json) :

```
      "env": [
        {
          "name": "DAPR_HTTP_PORT",
          "value": "3500"
        },
        {
          "name": "DAPR_GRPC_PORT",
          "value": "50001"
        }
      ],
```
But when daprd is the first container, there is NO env for daprd container.

And for the app container, the right env should be 👍 

```json
      "env": [
        {
          "name": "NAMESPACE",
          "value": "dapr-tests"
        },
        {
          "name": "DAPR_TRUST_ANCHORS",
          "value": "-----BEGIN CERTIFICATE-----\nMIIB2jCCAYGgAwIBAgIQVrWy6pqr7dv1USO+UgLZDjAKBggqhkjOPQQDAjAxMRcw\nFQYDVQQKEw5kYXByLmlvL3NlbnRyeTEWMBQGA1UEAxMNY2x1c3Rlci5sb2NhbDAe\nFw0yMTA1MTAxMjI1MTRaFw0yMjA1MTAxMjI1MTRaMDExFzAVBgNVBAoTDmRhcHIu\naW8vc2VudHJ5MRYwFAYDVQQDEw1jbHVzdGVyLmxvY2FsMFkwEwYHKoZIzj0CAQYI\nKoZIzj0DAQcDQgAERAUKTntckC7Q+SumkDG31+eGvLD9sjKEbRIYlFR6CBpQSZP9\n+0Ho5p7DCv1DRQpWQ74KfVyZG1oVcqtNqTjYRKN7MHkwDgYDVR0PAQH/BAQDAgIE\nMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAPBgNVHRMBAf8EBTADAQH/\nMB0GA1UdDgQWBBQNKJn9WfIormGzWtjZ7Mn5go4PRzAYBgNVHREEETAPgg1jbHVz\ndGVyLmxvY2FsMAoGCCqGSM49BAMCA0cAMEQCIFn63aU5BzgVyDmn/VJ8z7XMyQJ8\nngIHWZ9ISuL2WxfVAiB46Wmh8CoLOz3KmYFiWNVLJ/IusKLtqQ2L/d0RhGkL4Q==\n-----END CERTIFICATE-----\n"
        },
        {
          "name": "DAPR_CERT_CHAIN",
          "value": "-----BEGIN CERTIFICATE-----\nMIIBxTCCAWugAwIBAgIRAJ4JF93Dd4wKetJOBz++kOAwCgYIKoZIzj0EAwIwMTEX\nMBUGA1UEChMOZGFwci5pby9zZW50cnkxFjAUBgNVBAMTDWNsdXN0ZXIubG9jYWww\nHhcNMjEwNTEwMTIyNTE0WhcNMjIwNTEwMTIyNTE0WjAYMRYwFAYDVQQDEw1jbHVz\ndGVyLmxvY2FsMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEMHP4Bt3+j5KoKGdn\nPOact8vTWHBLx/7rB4BTubjNV7BXsBrmuXY6kvKeER5tfbdZBCPNzRbFU/eBy+1i\ntXC7aKN9MHswDgYDVR0PAQH/BAQDAgEGMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0O\nBBYEFJ1mMEdflNBXH/b3BUE6rdf16HFJMB8GA1UdIwQYMBaAFA0omf1Z8iiuYbNa\n2NnsyfmCjg9HMBgGA1UdEQQRMA+CDWNsdXN0ZXIubG9jYWwwCgYIKoZIzj0EAwID\nSAAwRQIgIH4F5OUUef61qjEgzikQ2am1/lQLZrnZEDIOAa8c0CYCIQC0e7mUkWZc\nwp4GWkSBMTrZMR7Hwcc/e2/rCTLqK21h7w==\n-----END CERTIFICATE-----\n"
        },
        {
          "name": "DAPR_CERT_KEY",
          "value": "-----BEGIN EC PRIVATE KEY-----\nMHcCAQEEIF0pM6TJUI0Qb4a0l7Po7+H+N+U/INaiXXOUddHFlK5IoAoGCCqGSM49\nAwEHoUQDQgAEMHP4Bt3+j5KoKGdnPOact8vTWHBLx/7rB4BTubjNV7BXsBrmuXY6\nkvKeER5tfbdZBCPNzRbFU/eBy+1itXC7aA==\n-----END EC PRIVATE KEY-----\n"
        },
        {
          "name": "SENTRY_LOCAL_IDENTITY",
          "value": "dapr-tests:default"
        }
      ],
```

But when daprd is the first container, the env of daprd container is 👍 

```yaml
      "env": [
        {
          "name": "DAPR_HTTP_PORT",
          "value": "3500"
        },
        {
          "name": "DAPR_GRPC_PORT",
          "value": "50001"
        }
      ],
```

Reply:
@yaron2 I have found the root casue:

```
for i, container := range containers {
		path := fmt.Sprintf("%s/%d/env", containersPath, i)
	}
```

Since daprd is added as first container (index=0), then the path of others container should be i + 1, not i.

I submit another PR for this fixing:

https://github.com/dapr/dapr/pull/3150

Reply:
@skyao #3150 is now merged.